samples/kvssink_intermittent_sample.cpp

```diff
@@ -0,0 +1,322 @@
+    g_object_set(G_OBJECT(clock_overlay),"time-format", "%a %B %d, %Y %I:%M:%S %p", NULL);
+
+    /* video convert */
+    video_convert = gst_element_factory_make("videoconvert", "video_convert");
+
+    /* source filter */
+    source_filter = gst_element_factory_make("capsfilter", "source-filter");
+    source_caps = gst_caps_new_simple("video/x-raw", "format", G_TYPE_STRING, "I420", NULL);
+    g_object_set(G_OBJECT(source_filter), "caps", source_caps, NULL);
+    gst_caps_unref(source_caps);
+
+    /* encoder */
+    encoder = gst_element_factory_make("x264enc", "encoder");
+    g_object_set(G_OBJECT(encoder),
+                 "bframes", 0,
+                 "key-int-max", 120, NULL);
+
+    /* sink filter */
+    sink_filter = gst_element_factory_make("capsfilter", "sink-filter");
+    sink_caps = gst_caps_new_simple("video/x-h264",
+                                    "stream-format", G_TYPE_STRING, "avc",
+                                    "alignment", G_TYPE_STRING, "au",
+                                    NULL);
+    g_object_set(G_OBJECT(sink_filter), "caps", sink_caps, NULL);
+    gst_caps_unref(sink_caps);
+
+    /* kvssink */
+    kvssink = gst_element_factory_make("kvssink", "kvssink");
+    if (IS_EMPTY_STRING(stream_name)) {
+        LOG_INFO("No stream name specified, using default kvssink stream name.")
+    } else {
+        g_object_set(G_OBJECT(kvssink), "stream-name", stream_name, NULL);
+    }
+    determine_aws_credentials(kvssink, stream_name);
+    
+
+    /* Check that GStreamer elements were all successfully created */
+
+    if(!kvssink) {
+        LOG_ERROR("[KVS sample] Failed to create kvssink element");
+        return -1;
+    }
+
+    if(!pipeline || !source || !clock_overlay || !video_convert || !source_filter || !encoder || !sink_filter) {
+        LOG_ERROR("[KVS sample] Not all GStreamer elements could be created.");
+        return -1;
+    }
+
+    // Populate data struct.
+    customData.main_loop = main_loop;
+    customData.pipeline = pipeline;
+
+    // Add a message handler.
+    bus = gst_pipeline_get_bus(GST_PIPELINE(pipeline));
+    bus_watch_id = gst_bus_add_watch(bus, bus_call, &customData);
+    gst_object_unref(bus);
+
+    // Add elements into the pipeline.
+    gst_bin_add_many(GST_BIN(pipeline),
+                        source, clock_overlay, video_convert, source_filter, encoder, sink_filter, kvssink, NULL);
+
+    // Link the elements together.
+    if(!gst_element_link_many(source, clock_overlay, video_convert, source_filter, encoder, sink_filter, kvssink, NULL)) {
+        LOG_ERROR("[KVS sample] Elements could not be linked");
+        gst_object_unref(pipeline);
+        return -1;
+    }
+
+    // Set the pipeline to playing state.
+    LOG_INFO("[KVS sample] Starting stream to KVS for " << KVS_INTERMITTENT_PLAYING_INTERVAL_SECONDS << " seconds");
+    gst_element_set_state(pipeline, GST_STATE_PLAYING);
```

Comment:
suggest checking `GstStateChangeReturn`  if itdoes not return `GST_STATE_CHANGE_FAILURE`